noxfile.py

```diff
@@ -0,0 +1,82 @@
+import tempfile
+from typing import Any
+
+import nox
+from nox.sessions import Session
+
+package = "session", "data_transfer"
+nox.options.sessions = "black", "lint", "mypy", "tests", "isort"
+locations = "consumer", "data_transfer", "tests", "noxfile.py", "cli.py"
+
+
+def install_with_constraints(session: Session, *args: str, **kwargs: Any) -> None:
+    """A wrapper for session.install use linting and
+    test depenencies that are pinned. This ensure
+    replicatability amongst developers."""
+    with tempfile.NamedTemporaryFile() as requirements:
+        session.run(
+            "poetry",
+            "export",
+            "--dev",
+            "--without-hashes",
+            "--format=requirements.txt",
+            f"--output={requirements.name}",
+            external=True,
+        )
+        session.install(f"--constraint={requirements.name}", *args, **kwargs)
+
+
+@nox.session(python=["3.8"])
+def black(session: Session) -> None:
+    """Automatic format code following black codestyle:
+    https://github.com/psf/black
+    """
+    args = session.posargs or locations
+    install_with_constraints(session, "black")
+    session.run("black", *args)
+
+
+@nox.session(python=["3.8"])
+def isort(session: Session) -> None:
+    """Automatic order import statements"""
+    args = session.posargs or locations
+    install_with_constraints(session, "isort")
+    session.run("isort", *args)
+
+
+@nox.session(python=["3.8"])
+def mypy(session: Session) -> None:
+    args = session.posargs or locations
+    install_with_constraints(session, "mypy")
+    session.run("mypy", *args)
+
+
+@nox.session(python=["3.8"])
+def lint(session: Session) -> None:
+    """Provide lint warnings to help enforce style guide."""
+    args = session.posargs or locations
+    install_with_constraints(
+        session,
+        "flake8",
+        "flake8-aaa",
+        "flake8-bandit",
+        "flake8-black",
+        "flake8-bugbear",
+    )
+    session.run("flake8", *args)
+
+
+@nox.session(python=["3.8"])
+def tests(session: Session) -> None:
+    """Setup for automated testing with pytest"""
+    from cli import run_command
+
+    run_command("pytest")
```

Comment:
Note: this is always a success regardless of return from pytest since it is not associated with the nox session (so cannot determine the value). For example, run:

```shell
poetry run nox -rs tests 
```

and then run `pytest` from the command line and note that 5 tests failed as `if config.access_authenticated_endpoints` is disabled.

Potential solution: 

change this code to `session.run("pytest", "-vs")` though not sure if that works for you @davidverweij

Reply:
@davidverweij note: if `session.run` does not work for you locally, we could alternative use the result of `run_command` to notify if nox failed, e.g.,:

```python
@nox.session(python=["3.8"])
def tests(session: Session) -> None:
    """Setup for automated testing with pytest"""
    from cli import run_command
    result = run_command("pytest")
    if result.returncode:
        session.error("failed")
```

Reply:
Good catch!

It doesn't work, but `poetry run pytest -vs` works. Hence, temporary fix:
```python
@nox.session(python=["3.8"])
def tests(session: Session) -> None:
    """Setup for automated testing with pytest"""
    session.run("poetry", "run", "pytest", "-vs")
```

Reply:
Can you test if that works for you @jawrainey?